src/test/java/com/es/phoneshop/web/ProductListPageServletTest.java

```diff
@@ -37,5 +41,6 @@ public void testDoGet() throws ServletException, IOException {
         servlet.doGet(request, response);
 
         verify(requestDispatcher).forward(request, response);
+        verify(request).setAttribute(eq("products"), any());
```

Comment:
Try to inject mock ProductDao into tested Object, to verify that `setAttribute()` called with what returned from `ProductDao.findProducts()`. You might need to update servlet class for that